lib/definition/component-info.ts

```diff
@@ -0,0 +1,103 @@
+import { ComponentsDefinition, Component, ComponentRendition } from '../models/components-definition';
+import {
+    ComponentSetFields,
+    ComponentField,
+    ComponentSetInfo,
+    ComponentInfoFields,
+} from '../models/component-set-fields';
+import parse5 = require('parse5');
+
+const directivePrefix = 'doc-';
+
+/**
+ * Creates partial info definitions of the component set.
+ *
+ * This information is used by the digital to print conversion.
+ */
+export function processFields(componentsDefinition: ComponentsDefinition): ComponentSetFields {
+    return componentsDefinition.components.reduce((result, component) => {
+        result[component.name] = {
+            label: component.label as string,
+            iconUrl: component.icon,
+            fields: parseFields((<{ html: string }>component.renditions)[ComponentRendition.HTML]),
+        };
+        result[component.name].fields.forEach((f) => addRestrictChildrenInfo(componentsDefinition, component, f));
+        return result;
+    }, {} as ComponentSetFields);
+}
+
+/**
+ * Creates partial info definitions of the component set.
+ *
+ * This information is provided with custom channel publish messages. Integrators can use
+ * the info to interpret the given digital article without additional API requests.
+ */
+export function processInfo(componentsDefinition: ComponentsDefinition): ComponentSetInfo {
+    return {
+        components: componentsDefinition.components.reduce((result, component) => {
+            result[component.name] = {
+                fields: parseFields((<{ html: string }>component.renditions)[ComponentRendition.HTML]),
```

Comment:
Yes, I figured that out indeed. The main problem in the types right now is that a lot of fields are optional right now, including the `renditions`. As it is hard for me to figure out why this is the case I solved it this way. Also because we use this rule: https://github.com/typescript-eslint/typescript-eslint/blob/v4.16.1/packages/eslint-plugin/docs/rules/no-non-null-assertion.md I'm not exactly sure how to solve this.